package com.itssvkv.chatapp.utilsimport java.io.IOExceptionimport java.net.SocketTimeoutExceptionsealed class CallResult<out T> {    data class CallSuccess<T>(val data: T) : CallResult<T>()    data class CallFailure(val msg: String, val code: Int) : CallResult<Nothing>()}sealed class CallState {    object EmptyState : CallState()    object LoadingState : CallState()    data class SuccessState<T>(val data: T) : CallState()    data class FailureState(val msg: String? = null, val code: Int? = null) : CallState()}suspend fun <T> safeApiCall(firebaseCall: suspend () -> T): CallResult<T> {    return try {        CallResult.CallSuccess(firebaseCall.invoke())    } catch (throwable: Throwable) {        when (throwable) {            is SocketTimeoutException -> CallResult.CallFailure("Time is out", -1)            is IOException -> CallResult.CallFailure("Check the internet", -2)            else -> {                CallResult.CallFailure(throwable.message.toString(), -4)            }        }    }}